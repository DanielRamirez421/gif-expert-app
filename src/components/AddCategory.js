import React, { Fragment, useState } from 'react'
import { PropTypes } from 'prop-types';

export const AddCategory = ({ setCategories }) => {

    const [inputState, setInputState] = useState('')

    const handleInputChange = (e) => {
        setInputState(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        
        if ( inputState.trim().length > 2 ) {
            setCategories( cats => [ inputState, ...cats ] );
            setInputState( '' )
        }
    }

    return (
        <Fragment>
            <form onSubmit={ handleSubmit }>
                <input 
                    type="text"
                    value={ inputState }
                    onChange={ handleInputChange }/>
            </form>
        </Fragment>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
