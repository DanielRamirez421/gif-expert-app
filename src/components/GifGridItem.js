import React from 'react'

export const GifGridItem = ( {id, title, url} ) => {

    return (
        <div className='card'>
            <div className='card--image'>
                <img src={ url } alt={ title } />
            </div>
            <p><i>{ title }</i></p>
        </div>
    )
}
