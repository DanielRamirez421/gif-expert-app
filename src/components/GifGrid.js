import React, { Fragment } from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category }) => {

    const { data:images, loading } = useFetchGifs( category );

    return (
        <Fragment>
            <h3>{ category }</h3>

            <p className='not-results'>{ loading && 'Loading ...' }</p>
            <p className='not-results'>{ (!images.length && !loading) && (`Not results for: ${ category }`) }</p>

            <div className='card-grid'>
                {
                    images.map((img) => 
                        <GifGridItem
                            key={ img.id } 
                            { ...img }
                        />
                    )
                }
            </div>
        </Fragment>
    )
}
