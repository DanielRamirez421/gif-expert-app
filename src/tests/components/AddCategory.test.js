import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('AddCategory component file test', () => {
   
    const setCategories = jest.fn();
    let wrapper = shallow( <AddCategory setCategories={ setCategories }/> );

    beforeEach(() => {
        jest.clearAllMocks(); // clean simulations mocks
        wrapper = shallow( <AddCategory setCategories={ setCategories }/> );
    });

    test('should show AddCategory component correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should change input element', () => {
        const input = wrapper.find('input');
        const value = 'Hola Mundo!';
        input.simulate('change', { target: { value } });
        expect( wrapper.find('p').text().trim() ).toBe( value );
    });

    test('Should not call onSubmit method', () => {
        wrapper.find('form').simulate('submit', { preventDefault(){} });    
        expect(setCategories).not.toHaveBeenCalled();
    });

    test('should call setCategories and clear input', () => {
        const value = 'One punch';
        wrapper.find('input').simulate('change', { target: { value } });
        wrapper.find('form').simulate ('submit', { preventDefault(){} });
        expect(setCategories).toBeCalled();
        expect( wrapper.find('input').prop('value') ).toBe('');
        expect( setCategories ).toBeCalledWith( expect.any(Function) );
        // NOTE: Don't encapsule wrapper.find in a variable!!!!!!!!!
    });

});
