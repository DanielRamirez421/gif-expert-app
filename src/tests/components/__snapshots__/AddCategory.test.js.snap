// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory component file test should show AddCategory component correctly 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
